### PR TITLE
route: support for instrumentation

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: route/route.go [hunk 1]
```diff
@@ -33,9 +34,22 @@ func New() *Router {
...
+}
+
 // WithPrefix returns a router that prefixes all registered routes with prefix.
 func (r *Router) WithPrefix(prefix string) *Router {
-	return &Router{rtr: r.rtr, prefix: r.prefix + prefix}
+	return &Router{rtr: r.rtr, prefix: r.prefix + prefix, instrh: r.instrh}
+}
+
+// InstrumentHandler returns the handler wrapped by the instrumentation handler.
+func (r *Router) InstrumentHandler(handlerName string, handler http.HandlerFunc) http.HandlerFunc {
```
Comment (brian-brazil) [inline route/route.go]: This shouldn't be exposed in the API, instead use it within the class

Comment (simonpasquier) [reply route/route.go]: Do you mean wrapping with the instrumentation handler every time a route is registered (eg in `Get()`, `Post()`, ...) using `path` as the `handler` label value?

Comment (brian-brazil) [reply route/route.go]: Yes, that's one way.

**Comment by simonpasquier** [comment]
@brian-brazil please take a look.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**